docs/proposals/002-api-proposal/proposal.md

```diff
@@ -0,0 +1,254 @@
+  - LLMUseCase allows for traffic splitting between adapters _in the same pool_ to allow for new LoRA adapter versions to be easily rolled out 
+- SLO objectives for the UseCAse
+- The Pools this UseCase is relevant to 
+
+### Spec
+
+**BackendPool**
+```golang
+// A grouping of model serving instances that are expected to serve the same model(s) and speak the same API protocol
+// The LLMBackendPool is also the scope for enforcing priority and fairness across different use cases.
+// When used with the Gateway API, the LLMBackendPool can serve as the BackendRefs of an HTTPRoute.
+// Generally the LLMBackendPool is owned by the "Inference Platform Admin" persona.
+type LLMBackendPool struct {
+        metav1.ObjectMeta
+        metav1.TypeMeta
+        Spec LLMBackendSpec
+}
+
+type LLMBackendPoolSpec struct {
+        // Pod selector, similar to k8s Service.
+        Selector map[string]string `json:"selector,omitempty"`
+        // Allows names within the `model` param that don't have an explicitly defined UseCase to pass through. Defaults to `false`.
+        AllowUndefinedModels bool
+        // Admin-defined minimum objective value that can be requested, will cause an error in LLMUseCase if limit is broken.
+        MinTPOT float32
+}
+```
+
+
+**LLMUseCase**
+```golang
+// LLMUseCase represents a use case of an LLM, which is multiplexed onto one or more LLMBackendPools.
+// It allows mapping a use case to backend pools, and models or adapters that backend model servers understand.
+// The LLMUseCase defines request routing behavior within an LLMBackendPool.
+// Generally the LLMUseCase is owned by the "LLM Use Case Owner" persona, which can be teams in an organization.
+type LLMUseCase struct {
+        metav1.ObjectMeta
+        metav1.TypeMeta
+
+        Spec LLMUseCaseSpec
+}
+
+type LLMUseCaseSpec struct {
+        // Map use case to one or more backend pools.
+        // In the most common case, this should be a single backend pool.
+        // Multiple backend pools can be used for traffic splitting while migrating to a new backend pool.
+        Rules   []LLMUseCaseRule
+}
+
+// LLMUseCaseRule represents a mapping from a LLMUseCase to a backend pool and adapters/models in that pool.
+type LLMUseCaseRule struct {
+        // The name used in the `model` param of incoming requests
+        ModelName string
+        // Optional
+        Objective *Objective
+        // Required.
+        // Reference to an LLMBackendPool. This allows registering a use case
+        // as valid on a pool. 
+        // NOTE: Allowing multiple pools is a configuration convenience.
+        PoolRef []corev1.ObjectReference
+        // Optional.
+	    // Allow multiple versions of a model for traffic splitting. 
+	    // If not specified, the target model name is defaulted to the 
+        // modelName parameter.
+        TargetModels []common.TargetModel
+}
+
+
+// TargetModel represents a deployed model or a LoRA adapter.
+type TargetModel struct {
```

Comment:
Do we need a separate field for model version?

Reply:
I think we should. Indeed, the target model can have multiple versions running.

Reply:
Our thinking was that the user could/would pack the version naming into the `targetModelName`. Would we prefer semantic versioning?

Reply:
I do not think we should impose a versioning on models, because we'll be having to deal with versioning defined outside of our domain (i.e. model names defined by others that users want to abstract).